sdk/src/types/block/output/account.rs

```diff
@@ -395,6 +397,40 @@ impl AccountOutput {
         self.features().staking().is_some() && next_state.map_or(true, |o| o.features().staking().is_none())
     }
 
+    /// Returns all the mana held by the output, which is potential + stored, all decayed.
+    pub fn available_mana(
+        &self,
+        protocol_parameters: &ProtocolParameters,
+        creation_index: SlotIndex,
+        target_index: SlotIndex,
+    ) -> Result<u64, Error> {
+        let decayed_mana = self.decayed_mana(protocol_parameters, creation_index, target_index)?;
+
+        decayed_mana
+            .stored
+            .checked_add(decayed_mana.potential)
+            .ok_or(Error::ConsumedManaOverflow)
+    }
+
+    /// Returns the decayed stored and potential mana of the output.
+    pub fn decayed_mana(
+        &self,
+        protocol_parameters: &ProtocolParameters,
+        creation_index: SlotIndex,
+        target_index: SlotIndex,
+    ) -> Result<DecayedMana, Error> {
+        let min_deposit = self.minimum_amount(protocol_parameters.storage_score_parameters());
+        let generation_amount = self.amount().saturating_sub(min_deposit);
+        let stored_mana = protocol_parameters.mana_with_decay(self.mana(), creation_index, target_index)?;
+        let potential_mana =
+            protocol_parameters.generate_mana_with_decay(generation_amount, creation_index, target_index)?;
+
+        Ok(DecayedMana {
+            stored: stored_mana,
+            potential: potential_mana,
+        })
+    }
```

Comment:
I'm wondering if we should create a macro for these two. Does anyone else have an opinion on this?

Reply:
Btw this change alone would be good for an own small PR

Reply:
🤷‍♀️  I don't really feel like it's worth it

Reply:
agree with Chloé, don't think it's worth it, but you may want to create an investigate-issue for this if you feel we should have another look at it.

Reply:
Not that important :trollface: